comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go

```diff
@@ -195,8 +195,15 @@ func (ac *AutoConfig) serviceListening() {
 	}
 }
 
-func (ac *AutoConfig) writeConfigCheck(w http.ResponseWriter, _ *http.Request) {
-	configCheckResponse := ac.GetConfigCheck()
+func (ac *AutoConfig) writeConfigCheck(w http.ResponseWriter, r *http.Request) {
+	raw := r.URL.Query().Get("raw") == "true"
```

Comment:
r is a pointer. Should we add a nil check?

Reply:
The `writeConfigCheck` function is called by the HTTP mux 

https://github.com/DataDog/datadog-agent/blob/ac6df4bac214e2ff6a242c6315418363f34abfe5/comp/api/api/apiimpl/internal/agent/agent.go#L53

I doubt that `r` would be nil, but adding the extra check would not hurt

Reply:
Done https://github.com/DataDog/datadog-agent/pull/27568/commits/af72f5f1af8caa20c6fd1a8d96c48e4844dab594